bindings/python/scripts/generate.py

```diff
@@ -1,105 +1,417 @@
+        self._linelist.append(f"namespace {self.name}" + "{")
+
+    def handle_struct_decl(self):
+        """
+        Handles `CursorKind.STRUCT_DECL` and `CursorKind.CLASS_DECL`
+
+        - Functions performed:
+            1. Define struct/class declaration:
+                1.1. Handles type references for templated classes.
+                1.2. Handles base specifiers (parent classes).
+            2. Handles anonymous field declarations (extract fields and declare as members).
+            3. Handles field declarations.
+            4. Handles class methods.
+        """
+
+        class_name = self.name
+        for sub_item in self.members:
+            if sub_item["kind"] == "TYPE_REF":
+                # TODO: Will this case only apply to templates?
+                # @TODO: Make more robust
+                type_ref = sub_item["name"].replace("struct ", "").replace("pcl::", "")
+                class_name = f"{self.name}<{type_ref}>"
+
+        cxx_base_specifier_list = [
+            sub_item["name"]
+            for sub_item in self.members
+            if sub_item["kind"] == "CXX_BASE_SPECIFIER"
+        ]
+        if cxx_base_specifier_list:
+            cxx_base_specifier_list = ",".join(cxx_base_specifier_list)
+            cxx_base_specifier_list = cxx_base_specifier_list.replace(
+                "struct ", ""
+            ).replace("pcl::", "")
+            self._linelist.append(
+                f'py::class_<{class_name}, {cxx_base_specifier_list}>(m, "{class_name}")'
             )
-            break
-    for line in module_linelist:
-        linelist.append(line)
-    linelist.append("}")
-    return linelist
+        else:
+            self._linelist.append(f'py::class_<{class_name}>(m, "{class_name}")')
 
+        # default constructor
+        self._linelist.append(".def(py::init<>())")
 
-def handle_include(item):
-    pass
+        # TODO: Merge this and next block via a design updation
+        # handle anonymous structs, etc. as field declarations
+        for sub_item in self.members:
+            fields = self.get_fields_from_anonymous(sub_item)
+            for field in fields:
+                if field["element_type"] == "ConstantArray":
+                    # TODO: FIX: readwrite, not readonly
+                    self._linelist.append(
+                        f'.def_property_readonly("{field["name"]}", []({self.name}& obj) {{return obj.{field["name"]}; }})'  # float[ ' + f'obj.{sub_item["name"]}' + '.size()];} )'
+                    )
+                else:
+                    self._linelist.append(
+                        f'.def_readwrite("{field["name"]}", &{self.name}::{field["name"]})'
+                    )
 
+        for sub_item in self.members:
 
-def handle_alias(item):
-    pass
+            # handle field declarations
+            if sub_item["kind"] == "FIELD_DECL":
+                if sub_item["element_type"] == "ConstantArray":
```

Comment:
Could you give an example of constant array?

_Originally posted by @kunaltyagi in https://github.com/divmadan/pcl/pull/8_

Reply:
A C-style array.
```cpp
#define PCL_ADD_UNION_POINT4D \
  union EIGEN_ALIGN16 { \
    float data[4]; \   <------------------------------------------- THIS
    struct { \
      float x; \
      float y; \
      float z; \
    }; \
  };
```

```json
              {
                  "depth": 4,
                  "line": 291,
                  "column": 5,
                  "kind": "FIELD_DECL",
                  "name": "data",
                  "element_type": "ConstantArray",
                  "access_specifier": "PUBLIC",
                  "members": [
                    {
                      "depth": 5,
                      "line": 291,
                      "column": 5,
                      "kind": "INTEGER_LITERAL",
                      "name": "",
                      "element_type": "Int",
                      "members": []
                    }
                  ]
                }
```

Reply:
It's read only (accessing it as an object) as of now, have to figure out how to write to it.

Reply:
Mark as todo somewhere

Reply:
Already marked.